src/openfermion/ops/_quadratic_hamiltonian.py

```diff
@@ -0,0 +1,104 @@
+    """Class for storing Hamiltonians that are quadratic in the fermionic
+    ladder operators. The operators stored in this class take the form::
+
+        \sum_{p, q} (M_{pq} - \mu \delta_{pq}) a^\dagger_p a_q
+        + 1 / 2 \sum_{p, q} (A_{pq} a^\dagger_p a^\dagger_q + h.c.)
+        + constant
+
+    where
+        M is a Hermitian n_qubits x n_qubits matrix.
+        A is an antisymmetric n_qubits x n_qubits matrix.
+        \mu is a float representing the chemical potential term.
+        \delta_{pq} is the Kronecker delta symbol.
+
+    We separate the chemical potential \mu from M so that we can use it
+    to adjust the expectation value of the total number of particles.
+
+    Attributes:
+        n_qubits: An int giving the number of qubits.
+        constant: A constant term in the operator given as a float.
+    """
+
+    def __init__(self, constant, hermitian_part,
+                 antisymmetric_part=None, chemical_potential=None):
+        """
+        Initialize the QuadraticHamiltonian class.
+
+        Args:
+            constant(float): A constant term in the operator.
+            hermitian_part(ndarray): The matrix M, which represents the
+                coefficients of the particle-number-conserving terms.
+                This is an n_qubits x n_qubits numpy array of complex numbers.
+            antisymmetric_part(ndarray): The matrix A, which represents the
+                coefficients of the non-particle-number-conserving terms.
+                This is an n_qubits x n_qubits numpy array of complex numbers.
+            chemical_potential(float): The chemical potential \mu.
+        """
+        n_qubits = hermitian_part.shape[0]
+
+        # Initialize combined Hermitian part
+        if not chemical_potential:
+            combined_hermitian_part = hermitian_part
+        else:
+            combined_hermitian_part = (
+                    hermitian_part -
+                    chemical_potential * numpy.eye(n_qubits))
+
+        # Initialize antisymmetric part
+        if antisymmetric_part is None:
+            antisymmetric_part = numpy.zeros((n_qubits, n_qubits), complex)
+
+        super(QuadraticHamiltonian, self).__init__(
+                constant,
+                {(1, 0): combined_hermitian_part,
+                 (1, 1): .5 * antisymmetric_part,
+                 (0, 0): -.5 * antisymmetric_part.conj()})
+        self.chemical_potential = chemical_potential
+
+    def combined_hermitian_part(self):
+        """Return the Hermitian part including the chemical potential."""
+        return self.n_body_tensors[1, 0]
+
+    def hermitian_part(self):
+        """Return the Hermitian part not including the chemical potential."""
+        hermitian_part = self.combined_hermitian_part()
+        if self.chemical_potential:
+            hermitian_part += (self.chemical_potential *
+                               numpy.eye(self.n_qubits))
+        return hermitian_part
+
+    def antisymmetric_part(self):
```

Comment:
I feel like since you call the other part the "Hermitian part" this should be called the "anti-Hermitian part". What do you think?

Reply:
I'm pretty sure the matrix really is antisymmetric (A = -A^T), rather than anti-Hermitian (A = -A^\dagger).